include/tsutil/Metrics.h

```diff
@@ -506,6 +507,18 @@ class Metrics
 
   }; // class Counter
 
+  class Derived
+  {
```

Comment:
Let's add doxygen comments explaining the responsibilities and purpose of this class.